org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterThermostatLocalTemperature.java

```diff
@@ -0,0 +1,124 @@
+            } else {
+                // Configure reporting
+                cluster.setLocalTemperatureReporting(REPORTING_PERIOD_DEFAULT_MIN, REPORTING_PERIOD_DEFAULT_MAX, 0.1);
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            logger.error("{}: Exception setting reporting ", endpoint.getIeeeAddress(), e);
+        }
+
+        // Add a listener, then request the status
+        cluster.addAttributeListener(this);
+
+        return true;
+    }
+
+    @Override
+    public void disposeConverter() {
+        cluster.removeAttributeListener(this);
+    }
+
+    @Override
+    public void handleRefresh() {
+        cluster.getLocalTemperature(0);
+    }
+
+    @Override
+    public Channel getChannel(ThingUID thingUID, ZigBeeEndpoint endpoint) {
+        ZclThermostatCluster cluster = (ZclThermostatCluster) endpoint.getInputCluster(ZclThermostatCluster.CLUSTER_ID);
+        if (cluster == null) {
+            logger.trace("{}: Thermostat cluster not found", endpoint.getIeeeAddress());
+            return null;
+        }
+
+        try {
+            if (!cluster.discoverAttributes(false).get()) {
+                // Device is not supporting attribute reporting - instead, just read the attributes
+                Integer capabilities = cluster.getLocalTemperature(Long.MAX_VALUE);
+                if (capabilities == null) {
+                    logger.trace("{}: Thermostat local temperature returned null", endpoint.getIeeeAddress());
+                    return null;
+                }
+            } else if (!cluster.isAttributeSupported(ZclThermostatCluster.ATTR_LOCALTEMPERATURE)) {
+                logger.trace("{}: Thermostat local temperature not supported", endpoint.getIeeeAddress());
+                return null;
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            logger.warn("{}: Exception discovering attributes in thermostat cluster", endpoint.getIeeeAddress(), e);
+        }
+
+        return ChannelBuilder
+                .create(createChannelUID(thingUID, endpoint,
+                        ZigBeeBindingConstants.CHANNEL_NAME_THERMOSTAT_LOCALTEMPERATURE),
+                        ZigBeeBindingConstants.ITEM_TYPE_NUMBER_TEMPERATURE)
+                .withType(ZigBeeBindingConstants.CHANNEL_THERMOSTAT_LOCALTEMPERATURE)
+                .withLabel(ZigBeeBindingConstants.CHANNEL_LABEL_THERMOSTAT_LOCALTEMPERATURE)
+                .withProperties(createProperties(endpoint)).build();
+    }
+
+    @Override
+    public void attributeUpdated(ZclAttribute attribute) {
+        logger.debug("{}: ZigBee attribute reports {}", endpoint.getIeeeAddress(), attribute);
+        if (attribute.getCluster() == ZclClusterType.THERMOSTAT
+                && attribute.getId() == ZclThermostatCluster.ATTR_LOCALTEMPERATURE) {
+            Integer value = (Integer) attribute.getLastValue();
```

Comment:
Suggestion: One could deal with the value `0x8000` here, which indicates an invalid temperature measurement. (But one could also leave this to future PRs...)